### PR TITLE
INN-2674: Add AppID as a field for state identifier

### PR DESCRIPTION
**Pull request by darwin67**
## Description

Adding the AppID as part of the identifier now that Apps are a 1st class citizen.

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: pkg/execution/debounce/debounce.go [hunk 1]
```diff
@@ -62,8 +62,10 @@ func init() {
 type DebounceItem struct {
 	// AccountID represents the account for the debounce item
 	AccountID uuid.UUID `json:"aID"`
-	// WorkspaceID represents the account for the debounce item
+	// WorkspaceID represents the workspace for the debounce item
 	WorkspaceID uuid.UUID `json:"wsID"`
+	// AppID represents the app for the debounce item
+	AppID uuid.UUID `json:"appID"`
```
Comment (tonyhb) [inline pkg/execution/debounce/debounce.go:68]: Why is this needed in debounce?

Comment (darwin67) [reply pkg/execution/debounce/debounce.go:68]: It's not necessary, just consistency.
I can remove it if you want.

Comment (darwin67) [reply pkg/execution/debounce/debounce.go:68]: Because it generates an Identifier here.
https://github.com/inngest/inngest/pull/1082/files#diff-32bae9a1a6a74b89472b6118cdb59253bd70bacfa0449bd221a87919678a64f6R228

I was basically updating everything that needs to create an Identifier.

**Review by tonyhb — APPROVED** [review]
